packages/frontend/src/components/create-campaign/form/restrictions-step/csv-addresses-import/index.tsx

```diff
@@ -0,0 +1,116 @@
+import classNames from "classnames";
+import { Typography } from "@metrom-xyz/ui";
+import { useEffect, useRef, useState, type ChangeEvent } from "react";
+import { isAddress, type Address } from "viem";
+import Papa from "papaparse";
+import { toast } from "sonner";
+import { useTranslations } from "next-intl";
+import { MAXIMUM_RESTRICTIONS } from "..";
+import { ImportFail } from "./notifications/import-fail";
+import { ImportSuccess } from "./notifications/import-success";
+
+import styles from "./styles.module.css";
+
+interface CsvAddressesImportProps {
+    onImport: (value: Address[]) => void;
+    className?: string;
+}
+
+export function CsvAddressesImport({
+    onImport,
+    className,
+}: CsvAddressesImportProps) {
+    const t = useTranslations("newCampaign.form.restrictions.import");
+    const [csv, setCsv] = useState<File | null>(null);
+    const [addresses, setAddresses] = useState<string[]>([]);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+        if (!csv) return;
+
+        Papa.parse(csv, {
+            skipEmptyLines: true,
+            complete: (result) => {
+                const addresses = result.data.flat() as string[];
+                setAddresses(addresses);
+            },
+            error: (error) => {
+                console.error("Error parsing csv: ", error);
+            },
+        });
+    }, [csv]);
+
+    useEffect(() => {
+        if (addresses.length === 0 || !csv) return;
+
+        setCsv(null);
+        setAddresses([]);
+
+        if (addresses.some((address) => !isAddress(address))) {
+            toast.custom((toastId) => (
+                <ImportFail
+                    toastId={toastId}
+                    message={t("notification.fail.malformed")}
+                />
+            ));
+            return;
+        }
+        if (new Set(addresses).size !== addresses.length) {
```

Comment:
I don't think we necessarily need to fail here. Can we just remove the duplicates? Wdyt?

Reply:
The reasoning behind failing in this case was to make sure the user knows there are duplicates in the file, not sure, we can also remove the duplicates.

Reply:
Addressed in a3f83c7fb8b545ebf03794983fefa21a70f5d482.